SC2093 about removing "exec" should trigger in loops

#### For bugs
- Rule Id (if any, e.g. SC1000): SC2093
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] I read the issue's wiki page, e.g. https://github.com/koalaman/shellcheck/wiki/SC2086
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related


#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
for file in rc.d/*
do
     exec "$file"
done
```

#### Here's what shellcheck currently says:

No issues


#### Here's what I wanted or expected to see:

SC2093: Remove "exec " if script should continue after this command.

ShellCheck does not emit this warning when the command is the last in its block, but it should still be triggering for loops.

--- a/src/ShellCheck/Analytics.hs
+++ b/src/ShellCheck/Analytics.hs
@@ -1468,17 +1468,18 @@ prop_checkSpuriousExec5 = verifyNot checkSpuriousExec "exec > file; cmd"
 prop_checkSpuriousExec6 = verify checkSpuriousExec "exec foo > file; cmd"
 prop_checkSpuriousExec7 = verifyNot checkSpuriousExec "exec file; echo failed; exit 3"
 prop_checkSpuriousExec8 = verifyNot checkSpuriousExec "exec {origout}>&1- >tmp.log 2>&1; bar"
+prop_checkSpuriousExec9 = verify checkSpuriousExec "for file in rc.d/*; do exec \"$file\"; done"
 checkSpuriousExec _ = doLists
   where
-    doLists (T_Script _ _ cmds) = doList cmds
-    doLists (T_BraceGroup _ cmds) = doList cmds
-    doLists (T_WhileExpression _ _ cmds) = doList cmds
-    doLists (T_UntilExpression _ _ cmds) = doList cmds
-    doLists (T_ForIn _ _ _ cmds) = doList cmds
-    doLists (T_ForArithmetic _ _ _ _ cmds) = doList cmds
+    doLists (T_Script _ _ cmds) = doList cmds False
+    doLists (T_BraceGroup _ cmds) = doList cmds False
+    doLists (T_WhileExpression _ _ cmds) = doList cmds True
+    doLists (T_UntilExpression _ _ cmds) = doList cmds True
+    doLists (T_ForIn _ _ _ cmds) = doList cmds True
+    doLists (T_ForArithmetic _ _ _ _ cmds) = doList cmds True
     doLists (T_IfExpression _ thens elses) = do
-        mapM_ (\(_, l) -> doList l) thens
-        doList elses
+        mapM_ (\(_, l) -> doList l False) thens
+        doList elses False
     doLists _ = return ()
 
     stripCleanup = reverse . dropWhile cleanup . reverse
@@ -1487,10 +1488,15 @@ checkSpuriousExec _ = doLists
     cleanup _ = False
 
     doList = doList' . stripCleanup
-    doList' t@(current:following:_) = do
+    -- The second parameter is True if we are in a loop
+    -- In that case we should emit the warning also if `exec' is the last statement
+    doList' t@(current:following:_) False = do
         commentIfExec current
-        doList (tail t)
-    doList' _ = return ()
+        doList (tail t) False
+    doList' (current:tail) True = do
+        commentIfExec current
+        doList tail True
+    doList' _ _ = return ()
 
     commentIfExec (T_Pipeline id _ list) =
       mapM_ commentIfExec $ take 1 list

